lms/djangoapps/mobile_api/mobile_platform.py

```diff
@@ -0,0 +1,61 @@
+"""
+Platform related operations for Mobile APP
+"""
+import abc
+import re
+from mobile_api.models import AppVersionConfig
+
+
+class MobilePlatform(object):
+    """
+    MobilePlatform class creates an instance of platform based on user agent and supports platform
+    related operations
+    """
+    __metaclass__ = abc.ABCMeta
+
+    name = None
+    version = None
+
+    @classmethod
+    def create_instance(cls, user_agent, user_agent_regex):
+        """ Returns Android platform instance if user_agent matches with USER_AGENT_REGEX for Android """
+        match = re.search(user_agent_regex, user_agent)
+        if match:
+            platform = cls()
+            platform.set_version(match.group('version'))
+            return platform
+
+    def set_version(self, version):
```

Comment:
This is not idiomatic Python. You should call this `__init__()` and pass in the parameter when creating the instance.